### PR TITLE
Fix countdown synchronized and infinite loop for the wait state

### PR DESCRIPTION
**Pull request by sravel**
Great job as usual!
I fixed a bug for the countdown. If the sound did not take 3 seconds, it was not synchronized.
I also added an infinite loop for the wait state, and the possibility to disable plugin through the menu

### CODE REVIEW

**Comment by werdeil** [comment]
Hi @sravel Thanks for this PR, I will have a look at it but I don't want to add a enable/disable option I don't think this is a good thing to have plugins installed but disabled.

So could please remove this part from the PR?

**Comment by sravel** [comment]
Yes I can.
and for infini loop? at wait state, I keep or not?

File: pibooth_sound_effects.py [hunk 1]
```diff
@@ -47,65 +47,66 @@ def pibooth_cleanup():
 
 #--- Wait State ---------------------------------------------------------------
 @pibooth.hookimpl
-def state_wait_enter(app):
+def state_wait_enter(app, cfg):
```
Comment (werdeil) [inline pibooth_sound_effects.py]: Remove cfg from all hooks signature when it is not necessary

File: pibooth_sound_effects.py [hunk 2]
```diff
@@ -47,65 +47,66 @@ def pibooth_cleanup():
...
         app.sounds["choose"].play()
 
 #--- Chosen State -------------------------------------------------------------
 @pibooth.hookimpl
-def state_chosen_enter(app):
+def state_chosen_enter(app, cfg):
     """Actions performed when application enter in Chosen state.
     """
-    if "chosen" in app.sounds:
+    if "chosen" in app.sounds :
```
Comment (werdeil) [inline pibooth_sound_effects.py]: Remove the extra space before the ":"

File: pibooth_sound_effects.py [hunk 3]
```diff
@@ -47,65 +47,66 @@ def pibooth_cleanup():
...
 #--- Preview State ------------------------------------------------------------
 @pibooth.hookimpl
-def state_preview_enter(app):
+def state_preview_enter(app, cfg):
     """Actions performed when application enter in Preview state.
     """
-    if "preview" in app.sounds:
-        app.sounds["preview"].play()
+    if "preview" in app.sounds :
+        app.sounds["preview"].play(loops=cfg.getint("WINDOW","preview_delay")-1)
```
Comment (werdeil) [inline pibooth_sound_effects.py:80]: Very nice way to do it.

**Review by werdeil — CHANGES_REQUESTED** [review]
Just small changes to do (things linked to the removal of the disable option) and we could merge it

File: pibooth_sound_effects.py (hunk 2 above)
Comment (sravel) [reply pibooth_sound_effects.py]: fix

File: pibooth_sound_effects.py (hunk 1 above)
Comment (sravel) [reply pibooth_sound_effects.py]: fix

File: pibooth_sound_effects.py (hunk 3 above)
Comment (sravel) [reply pibooth_sound_effects.py:80]: Thanks,

**Comment by werdeil** [comment]
I'm ok with all the fixes from my comments, thanks for the reactivity.

I misunderstood the title, I thought you were fixing an infinite loop in the wait state but in fact it is the contrary, you implemented it. 

This is an interesting point of view, I need to think a little if it is good to have a different behavior for the wait state.

**Pull request merged**